### ISSUE TITLE
Ore Dictionificator state isn't loaded on startup

### ISSUE THREAD

**Issue by mbainter** [mekanism/Mekanism#6333]
*Please use the search functionality before reporting an issue. Also take a look at the closed issues!*

#### Issue description:
When using the ore dictionificator the filter configuration only lasts until the next reboot.  It has to be reconfigured each time.

#### Steps to reproduce:
1. Place an oredictionificator
2. Setup a filter to convert ores
3. Reboot the minecraft server and check the configuration

#### Version (make sure you are on the latest version before reporting):
**Forge:** 31.2.21
**Mekanism:** 1.15.2-9.10.8.421
**Other relevant version:** 
* Minecrat 1.15.2
* Valhelsia 2 modpack, version 2.1.3

#### If a (crash)log is relevant for this issue, link it here: (_It's almost always relevant_)
There's no crash log, but there is an error that I see when Minecraft is starting up.
<details>
<summary>Mekanism error from startup process</summary>
<p>

```
[14Jul2020 23:35:10.226] [Server thread/ERROR] [net.minecraft.tileentity.TileEntity/]: Failed to load data for block entity mekanism:oredictionificator
java.lang.NoSuchMethodError: net.minecraft.util.ResourceLocation.func_217855_b(Ljava/lang/String;)Z
	at mekanism.common.util.NBTUtils.setResourceLocationIfPresent(NBTUtils.java:171) ~[?:?]
	at mekanism.common.tile.TileEntityOredictionificator$OredictionificatorFilter.read(TileEntityOredictionificator.java:278) ~[?:?]
	at mekanism.common.content.filter.BaseFilter.readFromNBT(BaseFilter.java:44) ~[?:?]
	at mekanism.common.tile.TileEntityOredictionificator.setConfigurationData(TileEntityOredictionificator.java:167) ~[?:?]
	at mekanism.common.tile.TileEntityOredictionificator.func_145839_a(TileEntityOredictionificator.java:147) ~[?:?]
	at net.minecraft.tileentity.TileEntity.lambda$create$1(TileEntity.java:88) ~[?:?]
	at java.util.Optional.map(Optional.java:215) ~[?:1.8.0_252]
	at net.minecraft.tileentity.TileEntity.func_203403_c(TileEntity.java:86) ~[?:?]
	at net.minecraft.world.chunk.storage.ChunkSerializer.func_222650_a(ChunkSerializer.java:395) ~[?:?]
	at net.minecraft.world.chunk.storage.ChunkSerializer.lambda$read$2(ChunkSerializer.java:132) ~[?:?]
	at net.minecraft.world.chunk.Chunk.func_217318_w(Chunk.java:456) ~[?:?]
	at net.minecraft.world.server.ChunkManager.lambda$null$25(ChunkManager.java:576) ~[?:?]
	at com.mojang.datafixers.util.Either.lambda$mapLeft$0(Either.java:162) ~[?:?]
	at com.mojang.datafixers.util.Either$Left.map(Either.java:38) ~[?:?]
	at com.mojang.datafixers.util.Either.mapLeft(Either.java:162) ~[?:?]
	at net.minecraft.world.server.ChunkManager.lambda$func_219200_b$26(ChunkManager.java:563) ~[?:?]
	at java.util.concurrent.CompletableFuture.uniApply(CompletableFuture.java:616) ~[?:1.8.0_252]
	at java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:591) ~[?:1.8.0_252]
	at java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:456) ~[?:1.8.0_252]
	at net.minecraft.world.chunk.ChunkTaskPriorityQueueSorter.func_219083_b(SourceFile:58) ~[?:?]
	at net.minecraft.util.concurrent.ThreadTaskExecutor.func_213166_h(SourceFile:144) [?:?]
	at net.minecraft.world.server.ServerChunkProvider$ChunkExecutor.func_213166_h(SourceFile:551) [?:?]
	at net.minecraft.util.concurrent.ThreadTaskExecutor.func_213168_p(SourceFile:118) [?:?]
	at net.minecraft.world.server.ServerChunkProvider$ChunkExecutor.func_213168_p(SourceFile:560) [?:?]
	at net.minecraft.util.concurrent.ThreadTaskExecutor.func_213161_c(SourceFile:127) [?:?]
	at net.minecraft.world.server.ServerChunkProvider.func_212849_a_(SourceFile:139) [?:?]
	at net.minecraft.world.World.func_217353_a(World.java:149) [?:?]
	at net.minecraft.world.IWorldReader.func_217348_a(IWorldReader.java:101) [?:?]
	at net.minecraft.world.World.func_212866_a_(World.java:145) [?:?]
	at net.minecraft.world.World.func_180495_p(World.java:361) [?:?]
	at net.minecraft.world.World.func_175651_c(World.java:949) [?:?]
	at net.minecraft.world.World.func_175640_z(World.java:954) [?:?]
	at com.refinedmods.refinedstorage.tile.config.RedstoneMode.isEnabled(RedstoneMode.java:20) [?:1.8.5]
	at com.refinedmods.refinedstorage.apiimpl.network.node.NetworkNode.isActive(NetworkNode.java:106) [?:1.8.5]
	at com.refinedmods.refinedstorage.apiimpl.network.node.NetworkNode.canUpdate(NetworkNode.java:110) [?:1.8.5]
	at com.refinedmods.refinedstorage.apiimpl.network.node.NetworkNode.update(NetworkNode.java:129) [?:1.8.5]
	at com.refinedmods.refinedstorage.apiimpl.network.node.ExternalStorageNetworkNode.update(ExternalStorageNetworkNode.java:87) [?:1.8.5]
	at com.refinedmods.refinedstorage.apiimpl.network.NetworkNodeListener.onWorldTick(NetworkNodeListener.java:26) [?:1.8.5]
	at net.minecraftforge.eventbus.ASMEventHandler_924_NetworkNodeListener_onWorldTick_WorldTickEvent.invoke(.dynamic) [?:?]
	at net.minecraftforge.eventbus.ASMEventHandler.invoke(ASMEventHandler.java:80) [eventbus-2.2.0-service.jar:?]
	at net.minecraftforge.eventbus.EventBus.post(EventBus.java:258) [eventbus-2.2.0-service.jar:?]
	at net.minecraftforge.fml.hooks.BasicEventHooks.onPostWorldTick(BasicEventHooks.java:99) [?:?]
	at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:855) [?:?]
	at net.minecraft.server.dedicated.DedicatedServer.func_71190_q(DedicatedServer.java:330) [?:?]
	at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:784) [?:?]
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:637) [?:?]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_252]
[14Jul2020 23:35:10.228] [Server thread/WARN] [net.minecraft.tileentity.TileEntity/]: Skipping BlockEntity with id mekanism:oredictionificator
[14Jul2020 23:35:11.191] [Server thread/WARN] [net.minecraft.tileentity.TileEntity/]: Skipping BlockEntity with id 
```

</p>
</details>

**Comment by pupnewfster** [comment]
Unable to reproduce this with V10

**Comment by XFactHD** [comment]
Is there any chance of this getting fixed for 1.15.2? This bug crashes a dedicated server when the Oredictionificator is placed with existing NBT data on the ItemStack. It is caused by `ResourceLocation#isResourceNameValid() / ResourceLocation#func_217855()` being marked client-only.
